src/app/containers/Blocks/index.jsx

```diff
@@ -19,7 +20,7 @@ const Blocks = ({ blocks, componentsToRender }) =>
 
     return (
       <Block
-        key={id}
+        key={uuid()}
```

Comment:
This will generate a new key after every render. This hurts performance because it prevents React from making optimisations to avoid re-rendering more than it needs to. The key needs to be consistent between renders.

I recently worked on an [issue](https://github.com/bbc/simorgh/issues/1489) to remove use of uuid in React components (or in my case it was nanoid, but basically it's the same thing).

You'll need to create the uuid and it add to the data on retrieval of the data. An example of this can be found [here](https://github.com/bbc/simorgh/blob/60e702f71aa9b1699b9e73a645ffb39b538d4768/src/app/lib/utilities/preprocessor/rules/addIdsToBlocks.js) and the implementation of it [here](https://github.com/bbc/simorgh/blob/60e702f71aa9b1699b9e73a645ffb39b538d4768/src/app/routes/getInitialData/article.js#L19). You can probably use the same preprocessor rule `addIdsToBlocks` to achieve this.

Let me know if you have any questions!

Reply:
thanks @jroebu14 that makes sense.